soroban-env-host/src/test/observe.rs

```diff
@@ -463,6 +455,67 @@ impl Host {
             0
         }
     }
+
+    fn auth_stack_size(&self) -> usize {
+        if let Ok(auth_mgr) = self.try_borrow_authorization_manager() {
+            auth_mgr.stack_size()
+        } else {
+            0
+        }
+    }
+    fn auth_stack_hash(&self) -> u64 {
+        if let Ok(auth_mgr) = self.try_borrow_authorization_manager() {
+            auth_mgr.stack_hash()
+        } else {
+            0
+        }
+    }
+    fn auth_trackers_hash_and_size(&self) -> (u64, usize) {
+        if let Ok(auth_mgr) = self.try_borrow_authorization_manager() {
+            auth_mgr.trackers_hash_and_size()
+        } else {
+            (0, 0)
+        }
+    }
+    fn context_stack_size(&self) -> usize {
+        if let Ok(frames) = self.try_borrow_context_stack() {
+            frames.len()
+        } else {
+            0
+        }
+    }
+    fn context_stack_hash(&self) -> u64 {
+        if let Ok(frames) = self.try_borrow_context_stack() {
+            hash_iter(frames.iter())
+        } else {
+            0
+        }
+    }
+    fn storage_footprint_size(&self) -> usize {
+        if let Ok(storage) = self.try_borrow_storage() {
+            storage.footprint.0.len()
+        } else {
+            0
+        }
+    }
+    fn storage_footprint_hash(&self) -> u64 {
```

Comment:
I wonder if we could enforce the footprint immutability invariant instead of logging it into observations. Technically there is no need for us to eyeball the diffs if testing framework does verification on its own.
That's probably not necessary for this PR, just an idea.

Reply:
Oh, because the tests should never be recording footprint either?

Reply:
The tests actually can/do record the footprint, but the recording behavior is not observable by the protocol. So removing this could reduce the noise a bit. In the enforcing mode we just expect the footprint to be immutable, so that would work better as assertion.

Reply:
Mm. I think as with recording mode for auth, I'd prefer to _notice_ when we change it. It's not especially hard to update and while it's not divergence-critical, it's unlikely we'll want to have the recording behavior change by accident.